lib/Viewer.js

```diff
@@ -185,16 +187,51 @@ Viewer.prototype.importXML = function(xml, done) {
       return done(err, parseWarnings);
     }
 
+    var diagram = null;
+    if (diagramId) {
+      diagram = find(definitions.diagrams, function(element) {
+        return element.id == diagramId;
+      });
+    }
+
     self.importDefinitions(definitions, function(err, importWarnings) {
       var allWarnings = [].concat(parseWarnings, importWarnings || []);
 
       self._emit('import.done', { error: err, warnings: allWarnings });
 
       done(err, allWarnings);
-    });
+    }, diagram);
   });
 };
 
+/**
+ * Switch to another diagram of the same XML
+ *
+ * @param {String} ID of the diagram to render (if null or non-existent, the first will be rendered)
```

Comment:
I am not sure if we should actually let to use a non-existent diagram ID and still notify the user a successful switch (or import). This should be a place to complete with an error as displaying the whole BPMN is not what user expects when they pass a diagram ID.